### PR TITLE
xcode 9 fixes and bug fixes from last update

### PR DESCRIPTION
**Pull request by mdyson**

### CODE REVIEW

File: FSQCellManifest/FSQCellManifest.h [hunk 1]
```diff
@@ -129,7 +129,7 @@ extern const NSInteger kRowIndexForFooterIndexPaths;
  Setting this property is equivalent to calling `setSectionRecords:selectionStrategy:`
  with a selection strategy of FSQViewReloadCellSelectionStrategyDeselectAll.
  */
-@property (nonatomic) NSArray<FSQSectionRecord *> *sectionRecords;
+@property (nonatomic, null_resettable) NSArray<FSQSectionRecord *> *sectionRecords;
```
Comment (zackdotcomputer) [inline FSQCellManifest/FSQCellManifest.h:132]: Does this correctly behave in implementation as `null_resettable`? I.E. will this property be `@[]` after you set it to `nil`?

File: FSQCellManifest/FSQCellManifest.m [hunk 2]
```diff
@@ -287,11 +287,11 @@ - (void)replaceSectionRecords:(NSArray<FSQSectionRecord *> *)sectionRecords
     }];
 }
 
-- (void)setSectionRecords:(NSArray *)sectionRecords selectionStrategy:(FSQViewReloadCellSelectionStrategy)selectionStrategy {
+- (void)setSectionRecords:(nullable NSArray *)sectionRecords selectionStrategy:(FSQViewReloadCellSelectionStrategy)selectionStrategy {
```
Comment (zackdotcomputer) [inline FSQCellManifest/FSQCellManifest.m]: generics?

File: FSQCellManifest/FSQCellManifest.m [hunk 3]
```diff
@@ -287,11 +287,11 @@ - (void)replaceSectionRecords:(NSArray<FSQSectionRecord *> *)sectionRecords
...
     [self replaceSectionRecords:sectionRecords selectionStrategy:selectionStrategy managedViewUpdates:nil];
 }
 
-- (void)setSectionRecords:(NSArray *)sectionRecords {
+- (void)setSectionRecords:(nullable NSArray *)sectionRecords {
```
Comment (zackdotcomputer) [inline FSQCellManifest/FSQCellManifest.m]: ditto

File: FSQCellManifest/FSQCellManifest.m [hunk 4]
```diff
@@ -1597,8 +1597,8 @@ - (nullable UICollectionView *)collectionView {
     return (UICollectionView *)self.managedView;
 }
 
-- (instancetype)initWithDelegate:(id)delegate
-                         plugins:(NSArray *)plugins
+- (instancetype)initWithDelegate:(nullable id)delegate
+                         plugins:(nullable NSArray *)plugins
```
Comment (zackdotcomputer) [inline FSQCellManifest/FSQCellManifest.m]: ditto

File: FSQCellManifest/FSQSectionRecord.h [hunk 5]
```diff
@@ -40,7 +40,7 @@ NS_ASSUME_NONNULL_BEGIN
  @note To update a FSQSectionRecord that is already on a FSQCellManifest,
  do not set this property. Instead, use the insertion/removal methods in FSQCellManifest.
  */
-@property (nonatomic, copy, nullable) NSArray<FSQCellRecord *> *cellRecords;
+@property (nonatomic, copy, null_resettable) NSArray<FSQCellRecord *> *cellRecords;
```
Comment (zackdotcomputer) [inline FSQCellManifest/FSQSectionRecord.h:43]: This one definitely behaves correctly though

**Review by zackdotcomputer — COMMENTED** [review]
Looks good, couple little nits/questions to address before merging though

File: FSQCellManifest/FSQCellManifest.h (hunk 1 above)
Comment (mdyson) [reply FSQCellManifest/FSQCellManifest.h:132]: Following the code is a bit ugly here but ultimately setSectionRecords calls: [self setSectionRecords: selectionStrategy]
which calls: [self replaceSectionRecords:sectionRecords selectionStrategy:selectionStrategy managedViewUpdates:nil]
which ends up going to this line: https://github.com/foursquare/FSQCellManifest/blob/master/FSQCellManifest/FSQCellManifest.m#L271

File: FSQCellManifest/FSQCellManifest.m (hunk 3 above)
Comment (mdyson) [reply FSQCellManifest/FSQCellManifest.m]: fixed

File: FSQCellManifest/FSQCellManifest.m (hunk 2 above)
Comment (mdyson) [reply FSQCellManifest/FSQCellManifest.m]: fixed

**Review by zackdotcomputer — APPROVED** [review]
lgtm now

**Pull request merged**